jte-kotlin/src/main/java/gg/jte/compiler/kotlin/KotlinCodeGenerator.java

```diff
@@ -0,0 +1,658 @@
+package gg.jte.compiler.kotlin;
+
+import gg.jte.ContentType;
+import gg.jte.TemplateConfig;
+import gg.jte.TemplateException;
+import gg.jte.compiler.*;
+import gg.jte.runtime.ClassInfo;
+import gg.jte.runtime.Constants;
+import gg.jte.runtime.DebugInfo;
+import gg.jte.runtime.TemplateType;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static gg.jte.runtime.Constants.TEXT_PART_BINARY;
+
+@SuppressWarnings("unused") // Used by TemplateCompiler
+public class KotlinCodeGenerator implements CodeGenerator {
+    private final TemplateCompiler compiler;
+    private final TemplateConfig config;
+    private final ConcurrentHashMap<String, List<ParamInfo>> paramOrder;
+    private final ClassInfo classInfo;
+    private final CodeBuilder kotlinCode = new CodeBuilder();
```

Comment:
BTW, did you consider using [KotlinPoet](https://square.github.io/kotlinpoet/) here?

Reply:
No, it would make life more complicated than it would help and introduce an unecessary external dependeny.

CodeBuilder keeps an internal mapping for template lines to generated Kotlin lines and can be used for both Java and Kotlin generation.

Reply:
Got it, thanks! :)